frontend/src/store/calendar/actions.ts

```diff
@@ -0,0 +1,23 @@
+import { Dispatch, GetState } from '../types'
+import { ActionTypes, CALENDAR_ACTIONS } from './types'
+
+export const SetIsClicked = (newClickedDate: number) => async (dispatch: Dispatch<ActionTypes>, getState: GetState) => {
+  const { isClicked, clickedDate } = getState().calendar
+  if (isClicked) {
+    if (newClickedDate !== clickedDate) {
+      return
+    } else {
+      dispatch({
+        type: CALENDAR_ACTIONS.SET_IS_CLICKED,
+        newIsClicked: false,
+        newClickedDate: 0,
+      })
+      return
+    }
+  }
+  dispatch({
+    type: CALENDAR_ACTIONS.SET_IS_CLICKED,
+    newIsClicked: true,
+    newClickedDate: newClickedDate,
+  })
+}
```

Comment:
if logic becomes more complicated, you can consider setting parameters newIsClicked, newClickedDate, then based on logic set them to different values.

So at the end it's clear that we are doing one thing, dispatching and "Set Is Clicked" (based on method name)

Reply:
isClicked parameter not used anymore